### DISCUSSION TITLE
Svelte @apply in imported component css file

### DISCUSSION THREAD

**Discussion by benjaminpreiss** [tailwindlabs/tailwindcss#12198] [Help] [answered]
So I want to create some component-specific classes in a `.css` file, that I import in my Svelte component:

```
/component
    Component.svelte
    styles.css
```

Within my `styles.css`, I want to use `@apply` to apply tailwind classes:

```css
.some-class {
    @apply px-2
}
```

And then import and use `.some-class` in my component:

```svelte
<script>
    import "./styles.css"
</script>

<div class="some-class" />
```

When I [read the docs](https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css), I understand that this should not be possible, as:

> ... Tailwind has no idea that the card class exists since Svelte processed Card.svelte and main.css in total isolation from each other.

I wonder now, why does it work in [this SvelteLab REPL](https://www.sveltelab.dev/?t=tailwind&files=.%2Fsrc%2Flib%2FCounter.svelte%2C.%2Fsrc%2Flib%2Fcounter.css#code=N4IgdA5glgLlEDsD2AnApiAXKAZlANhtiAMZIIxoUDOWoA1lAgCZYjUBua%2Bl%2BAhgCMAtAFcoADj4gANCAEicONCiwBtACwA2aZvHTxAZmkBOAKz6jARkvbrl6dfUOADPcvOH7z2%2Bcf39sxdjbxc-SwB2Fz1-T3MvdUjjaIiXOOdo3xdpBOzbSzjLDJ9k21dstMj3AtLsyssAJhNE4PdEyucrMvjSty9rDNyXXrC9LWzGrwMjMZjrEcHrc2N2loaooaz%2B1M8eweNV4ZcJ5zTp0rzzGZtt9xau-J2moM9G9UvjzwzqjbH9kLmjjcOgtrosQaU0ndet9ls9rK93i4ALoAXzRsjAAAd0DA4MoUCQ6CA8IQiWQKFQYLRiIwWGxONxeIJRBIpLJ5IplGoGp0TNIDE5rHU0vZxE5YUlYvynKY9AYJqDLIKUjF1IKPMEBVKqp4-MD3EV9L06r1dlrZfyFXlle17GqsprldanjqDTc%2BsDNCa4c5grDzXLGlqAXZdRstfavI6Xp51QUwtpxO00gmXE4A5bssF1KNRuq%2BcG1m6hS5OvGpRn5SZ7IWFerTY9XfcnG8Y8WVfqTmHW2CtrN0o2ylrjNN89H4esS%2B4y2HDBtYY4THcnLp3crzJXGsYay3tMPRw7pf9ChtF2eu%2B49dCjxaq9uHPUjFXg85jmlDWDN0f1K8Lp9TwO1hmi26ZBsqvRWu6hyfjKgYPk%2BYExguSras2N5wYWBQfoBoJDqBD4bhM9RLLyDRpKi6IgJifAkPQfAQGgYAAFbUOQRIkkQoDkpQNBErSrCYOwXA8NwzJiJIMhyAoSgqJgqg8oehYais4awZmmH-v2JQAXoEqjDCyaTpYdzlpe35jkefZob2F7WGGX4RqM2g5oM6jRpGGqWUWKGuieyHAdIt6QWeHZ2hZha2PUEGqYFGHqgcoSeO0hoRuF65LrG7rBQ8X4KVGBbiskhnFpkDnKl8g41luRVAhkHipYe46ZLMJ5lZptlmscV7-g1%2BXjt59btSKnjRkFXkVUB4YKnWp4nk2vTuRZTUKr08UutcbUTueDaTfNmVVq2PZFjh7QzgCR3Dch4rLuUiW%2BfhYwHZcBSrHGlWZo9IGNUefzuINv3OJEm0PFZxTvT2gPunuCpxFFYZ-M%2Bcpwb2c1dDhuytgD-13EDCK3btDy-XZIavkCCGue0f5E9BpNNmcLh-qGAIvnyZHZGld33Hoq4tccNPdQCAPDTBsWZnlnnRpo6Wg0ZuzuH%2BOQLpkyHBWhmN1HUeQ7qLz7ijKHhjC5htyl9fVHlL2qo2Du1owr7TFRtam6yYLbZtopgG05gWIR5fL9TNLV5BCs3pk7iHGPrgymM5zkm%2Bz321plgeDsLKFHccNXTu6BsFFhOnvW1EeBQbznZl7bnmQn0tW7LMVjcOMq-tk5dx77fpeXnWng0sN0xADQNhz9Mp2u7TemNrbcFUNBeDw33tRyPgWh-HZuJ0zIPXETufoepeuBTH8%2BG8Epg%2B0tXkrZb2hb1KEPJAE7Q5xl696m%2BpY787Rcn655eLVXlvYdpVUBQA73ADnZAG14Rb1z3qYemRtR6V1XuKDWjZiYDQAq-BoAsUKF1ds3KOY8Nymwll5K%2BxUcImTDGZa%2BUC4ou0CovCuYxYGWU8mzPoxFkRohRLITESBqAwBINQagYByR4AgKIli7ECCcVIOQHiVI%2BJMAEkJRkolhDiTZFJTkslVDUyoRvaGjwlbvhGtnRKNCcqNE0FVJO9wrAdgKINdQ9VVhkNqmY3yeoNSrG3gYbM9hTB3CfmgsI3w7woIJtFbagJXT1GwfhGxYsfL2KSmuH4rj1qa2PCAnwmD4kh35AEwKwSowJRDIYzM%2BiSaYOBFWJJVZxas2OpnTspkHhjXvGbHIOQ5xM19FNNsJ035eJvvhEqFRBmTRMUZShJMfGW0uHUbB3xRlZ3tI0j0PgfD1WyrYkKyyqnYQvjEqsYoXQXgoeYsyk83EXNMdYOZHoFko0uAbMK30emRE0HkbJmydrOEGiFS44zAKXNtFUyJipokpLyTcApLUvqbIFmjXZqD9k%2BRVJMLc7RVarRXiQ5C2T165NiReBF1kJakU4UEzYepynNQ7IaGYXQEXeXxQTQiOhbHVI9LUowtL3AUR4ewAkRJmBQHQCQGAqAACeRI%2BCYkxGAAAFjAAAtvgaRpJiDcUpNSBgyj6TCSZBo1kkkOQyTUJoeqRhVzhGCF6fQThxDBHEB4TQmpjhOh9NzD4Nq0yDjuMjAcyE9R2u1i1eqK4PheADX9ecD8dAfEdfS2a7i7w3SuqhBsDSbSDjxtHI8B1vkpp0Gm5O-oFTKR9PYPN08WqrBOZivc%2BFeWpijY8etjygQdsBdyzmvQKoQPxsZXtg5l4luTYeeNYLTKQ02UCgc3auztuLd6uxcL60GGFO1a2uEfAOwPshbeydPyRByO4HF0811XGrW-LUU7NBlp3f-fONtD1QwWP2vuSbX1xoNu0f6Sbn10vLSYO%2Bd04aNOVtk2EF092137Ee0dpL2z30ShDGt0zvmRp-Wgj8OCY0zu8e6ZGlSGjWzKCuvUp6Wkujhp08pfQD0GL-FqUD-6N4APeq0L9vxINmUo-yb5gHOZ51AwB2oEH-jNThpJvU4HL34dKMG5NwrZCKuVXwgRQiDXEhkWSeR%2BqlF0kEgyES-AzUSXZNJLkckLY4Y8T2tZpVqrrWGoKj1PqZntieaRzNsmm3HmwR2jpwTHOb3IckfziVNnEsuRNXCqdzBeayBaEKvQGyNMfIeQ6p7IaY10s0FwF60iDw9vg0wMpgnZmeqhMrMmJQ41uuPfG%2BHTFjTS3V0dkyitPH7rdCuY0E02VGJ7eOB05QXoemXfBr71CBNq0NgrjXJTNdK61jwI2OvrBcfgjZrxpuTdeHNlyC2lt5fqzZNbxWn6FdS4EkN0sBz7cqy5arJ2l7zwW-hUwkQi0XZKVd3rm3%2Bsbb462L%2BO2XvjeLlVpFjRodBhm5mf7DCRPZEu-lfLDXbrrZK5D1LArnuoTh%2B9vQn3Dvw-O6jqsRbadY%2BBzj67IMHu8bZc5S4x9SdjeyBN6njPBfHxrBe7Hnlcc3dHXLTnW2dYbr5-to2D1jvU8q6YDcYvmcS9Z311ZrLUxQ6e2LXbbo3vbaO19wHdOke28x4t7XfJJds8azbJdUIhs85N7D-nWZAoHgOmd1Xf2BVPi191lbeOjqlHdz8VL23eds-Jxbg7f29w2-nl-IHEfnd64nQBD3Rv5eBsV9tibX9hcHeD2j4e9vxdO912DtIzbUz12CA6xDIVr2vKKV99WbZ6NAO1HDFGcsBnU5%2BZOWyepIjhDaBlPpI%2BKNo3b2EaC1xqf%2BIsABBx2zMziHMOEJ82Z8LiAvpKdeWsvsd6wQ%2BHwnRDjU4PzJqcLHfkEr9joefFT3ExJKnmdSVfdJKtbfMNCsE-QAnQZIcCcdM5IwMAz8CA58dvPmIyQjYOP4EA8IQjCFLfanMEBAh4BDRFIEJHZbXPJvWJUJR4CGZjNfYvLfcIWjTKLA70CpGsJA44Og48YBAbD3NLdhRTC-YmOTfZXLPqO7QbLOTLNJeZfHBlQTFCEbPyOoFsR%2BYg9nAne7RrNrB3CPapHCfFZQu7RcJXRvInDnQ3bnJnCPW4dadUM6LwMaeBQKYnAvR3O4FvGg5vC5OZQCRjcjEwtOdQ1bdrLwyaPQ5bDbEdH-eQ2TRQweLYbNMw0HCwt3dwofJhevQQrIRzAZfsZgkA-bLaWFMlDlIgnueI48FsbeLGDIsoHbbDFJLXHLXkL3GQy-MWcQkhCgiwrQwbQ6L3GDH3JXAAqbe3WbebcPKIyPKXcHQndndXRPUxUYvvavQPfeOvDwhvVIzQyQlrIvGHMnX3ZXD6dY07H7aYz5cwvYvgyg6HZYvbAXFXCYtHMvLY-Q2Yl3fHfY%2B4knb3Y41YwXc4w%2BVXevduXo9nD9B8Kwv3I40vVPM414%2BnC9ME7YnrDQ13VjW-CI%2BPYvUbJPE4sYy3X7NHWOD4mYyErEmXWEtreEwkoEl40k%2BnGUMPGw5bHIiQ2TPwaDIiNov2BQ6o0HOo-sfgkNJo9UAHDkzhD0P2XGeCb6AwhI-ElQzKFIzE6XEvFfOPDHSIw8TkiWAHZIJLNIfXHZLo-kv0QU9eRI03V7d4z7DXPvIPZEpHNw7PaUtIAiLhSifAKAAQcVSVNAaVOVIkAAYSQBEApBQDAAs0oG1VkT1V4hpCNXMxNXURZBs20StQczCRCTehKO0AU1ZkuWCxtMeGdkNkJzQUOWhNMKvl8QrAshKmyy3GY1rI2msVsQ%2B1KU8gMHdhDR8IOheRrLY0zFAwaQuKNjEMtIuUIx9X22MAbOPA7KMV8i5WoVrM7HKONGOOE2GLJkDSmBMGcnBIFJXLHPqVsXR1sJlKEKoMOSJmDmfgBIQOajBTrWSm2AGPeQtMVPbMvNeAjQ%2BmPk9NyLEwINeiyi7VjQ9Q1EzkVCv2Zi8ItABmHRaySXFmQSoXfMNCdQmW0O9MCzdBPSlHrWELyDRQou7CWAyFaW1mLMdS-J1CkNRT-N5VXO9PygUg4sAoHSSUnL93qXJi3xtkhHDC9QYLuSbA90QK4rguk2EOJWhTDAUw0zkSjMoBjL0wTKMwpGTMNTM1UUszEnNVsx0TUAE1QTRSaXbgthViGWiF5KoPXy5xfxELqmSSIjtDxM2SIkfizzAV3LkIKO2JyKsoJkfgv2GRiFaO%2BnsrqGmknGcuiumFoqMghSh2mFzGSRpO8M8R-Q6g2FvK9Ldncsop%2BIyuKoyBgJlkXDirNgSoHwzjvyOTfn63bEytS0uBypKuFUohQEjMoH0wlSlRlRQHlWIAAGp%2BBZUhrYz0zdLdVjMDKQB%2BJjU1ErNMytFLV7NVB41CYoLfJIJGKgQyy0EfVkDXJnJv9vIm4SjIFN5Gy5K0sA1L1Dq7FeDJpiypM%2BdOin1jh1KprqIGJYzlAuAYypFiAOI9KFF9N1q0zNrTKszdrdFZg2Vm1Lq2yQtL4TdIMlY0VBLSLhNBZJKldbEBKgw%2BT9TSiCZVD7JqazZBYVIYg29nV7lZlrlSowKBCaU7ggaQbGI4zZEYblr9LFEUyjLhatrNELU7NdEDrKjzwvrrhTqZKB9yijF8LRzYiSCYMgrngrr%2Byp90qchtbgq39CkrhO40ZAEhxQL9SDcNak5Oyp9WyLE4MpTwLpNILPqB8izYLwME0K5iLQ1d0wRu1sTZci8cSBx9acbcT3KqMbKo7YT7wDAfCr0MdVQcqRxzYy1whUD-8QgGrYNjEL0KqlZZLOFjypNlM8EfqFLzbZDLa9bXhS0fTuFKIBE%2BA4BCRiBRrgzxrJrcA%2BAOAoByQsQEAIAlquIVqJbDKVFpbkadr5a1BHwWgulXV9APBsDpBwg3BeR6gagPBT6HAjAj997rE9Az9pAz6S477H7o1sh%2BQn7H6N6HAt63VzAP7jI3Bt6GhIh77%2BQjB3V97EwVxghgHf6v6dAnAPAYH-7HV4HpBEGHBt6LYEHRLYHMHUHsGkGdBzAsHN6CHcG0HChHUiG8GSH0HKH378HaG4H6GaGKGmHn7zlyGAGgG360FExGg965xiHP6CHOHghRGmGchxHBUTxxHZH%2BHUHZ97BgHgtC79BGyz6UJxGNGnBxH7BxH1ArAdHuGz7xHL7VxNBzBzlb9ZGGHWGbGWGuHqHhHGGXI1hxH7wlynG-7GH7HnHWGhHvHWG3HbHt7fHAnt62s4YwnYHLAjBomCHYnUG9ApH%2BGnwHFBR6ggwWhOF4mfGQmxH8n96CmHHim-HQnxQiJSg4nCmonCmTwf6anCmrBGh6gb62nPAHEWnd7tBdGx1dHMmvHYHIh%2BnqmSnBnSGWnCnPtjBGgAnYGaUqHxHxA605mRHGmxn9HpgBVhm6miCr7-dxmfG9H1mymxHKgCldGGmNm6n6gdHdmWxRnTnDm7G6m7mxmO8DH%2BR3iO8wGcTcs59nnQmTnwm1NcmXmNmiwb8AdUt%2BG3BUGLwDBjnrmxmUY3mnndGuxxHPVXDAXN7BQAXFxZHEmlRHmQXcXWqs5MLwtyWFJWwwWgWUW7JemkX0XCnJQnSjpIgrGCln96XgsZnyWMXSmyXhN9G61kmrnWWUXamxmxg%2BX4ct8opKhX15W5x9GSsa7tBVGln0xt4MWOynwVxSXYHPn5WzWM6HAexpG9Gx18huZH55XNBjWCHgHd67V29uZknCnJ9PHRH%2BqRVhbRFyBxFmJ9NRa57xb4bUzjLTVtq5aLK5J1aHqXa1z0FeYXaAhZKN4NzaZMUfBJLRLuN31WN05N0B8YCYjs3e8BDSznaQojECb1o8V9pjgDa7CrrIsbb90x87YLkmjm3TClgoUr8t8a3pMfxxhxyfUSbjoZyzq62W1fct9O3O90YAJe2%2BNfI9xikxhm6alDy2YDp26HRuDR8qlT3aZLlyZBLbLNst38TujP8vlIUPb-arah2W6R3wj-CMUChEqF3p2rFLZ82pl7h5h7oE78iwPUEQVklhYhwFdEKB9UtRgL17Z%2BNGYa52wMZShsFUO7pwFGzFYk0HK6y04P3X8v2XKW2ngYMTSaCagIS0PMxL6s6csssW7QV18ktaqPQs3UwYM-F71-qYYW6kryy6y3MhT2OLzNlU4g43a4svCIk4OhlM3SDvzJLO0rInFIP%2BOaDxkN4nllyE1-lP3cqtSeUSP-2yPdxPlmO4CAJ9kVNUFROLRn3N2irrbV27bF4hMuYfbGV1OwRiPX3CzClPxW3pOuChTYrME%2BVYuBl-IoIGaBtFC-lu8y7wuSCQvGO5SwvvI5YlC4JG3SPsTgimduLOFfZuK03QEM37tHqUOqkfBP8SigU03CPZpt5ijH325zrHqTOq3lDnQpw0Y2vqvWu-RqVUtezTwMbYvXb-O4UiVa5iLqCs66Ztd1KYA%2BACAAB3ZRINhAENkgKG3AQzMWuG0zJe9MmWsy7MvatdM99bmmflccudvqU98r1vErjgimS2a6cd14Y9idmYbo6dqDgA%2BWMZN9LuaEyMRUWGDcabj-WrmPeo43Uutrp2sj0jEbcQu81mar-MWrlqsbxwvMzpAIGbx3Pj7k2Qj879NwBbiTpHj789yD97rzdS8eygY7kN87gzHVCN67yW27pG6zVehN%2BSWt5N%2Btv8uTnz3jSGe9nLHNttgZDt6uHjSTjd1XrzuoJLM0B2j0cS%2BXxduKmDFFeL7qPdhLrb9Xk3TXgt1YY3iSungWVni6isrqDBLJNbv3-sW3jYe3l%2BQ83r-kraGKtpAxZru8R63X7Eg374cmkwWnvUoVLuoAA) ?

... My guess is, that imports that end on `.css` are processed by tailwind (?)

**Comment by wongjn** [comment, marked as answer]
I believe the difference is that you `import`ed the CSS file in the JavaScript `<script>`, whereas the docs are referencing CSS in the component styles defined in `<style>` tags. CSS files `import`ed in JavaScript are "global" and not component-scoped, thus Tailwind works in the SvelteLab REPL.

---

As an aside, you may wish to consider [avoiding premature abstraction](https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply) and also avoiding `@apply` altogether as per Adam Wathan, creator of Tailwind:
- https://twitter.com/adamwathan/status/1226511611592085504
- https://twitter.com/adamwathan/status/1559250403547652097
